### PR TITLE
Add classes

### PR DESCRIPTION
**Pull request by ImMPrada**

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  def next_set
+    @encrypter.add_score(@current_set.turn)
+    return end_game if target_score_reached? && @is_pair
+
+    change_roles
+    new_set
+  end
+
+  def end_game
+    winner = hwo_won
```
Comment (mapra99) [inline lib/game.rb]:
```suggestion
    winner = who_won
```

File: lib/game.rb [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    return @render.print_end_game(@player1, @player2) if winner == @player1
+
+    @render.print_end_game(@player2, @player1)
+  end
+
+  def target_score_reached?
+    @player1.score >= @score_limit || @player2.score >= @score_limit
+  end
+
+  def hwo_won
```
Comment (mapra99) [inline lib/game.rb]:
```suggestion
  def who_won
```

File: lib/game.rb [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    else
+      @encrypter = @player1
+      @player1.change_rol(:encrypter)
+    end
+  end
+
+  def change_hacker
+    if @hacker == @player2
+      @hacker = @player1
+      @player1.change_rol(:hacker)
```
Comment (mapra99) [inline lib/game.rb]: The correct english for rol is role
```suggestion
      @player1.change_role(:hacker)
```

File: bin/main [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env ruby
+
+require 'colorize'
+require 'tty-font'
+require 'byebug'
```
Comment (mapra99) [inline bin/main]: byebug should be removed

File: lib/player.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+class Player
+  attr_reader :name, :rol, :score
```
Comment (mapra99) [inline lib/player.rb]:
```suggestion
  attr_reader :name, :role, :score
```

File: lib/player.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+  def initialize(name)
+    @name = name
+    @rol = nil
```
Comment (mapra99) [inline lib/player.rb]:
```suggestion
    @role = nil
```

File: lib/set.rb [hunk 7]
```diff
@@ -0,0 +1,120 @@
+class Set
```
Comment (mapra99) [inline lib/set.rb]: Given your definition "A set represents one round of the game", I think `Round` or `GameRound` is a better name for this class

File: lib/game.rb [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+  def initialize(score_limit = 6)
+    @player1 = nil
+    @player2 = nil
+    @encrypter = nil
+    @hacker = nil
+    @current_set = nil
+    @sets = []
+    @render = Render.new
+    @score_limit = score_limit
+    @is_pair = true
```
Comment (mapra99) [inline lib/game.rb]: "pair" means "pareja", not "par". The right translation is "even" (and "odd" for "impar")
```suggestion
    @is_even_round = true
```

Also I think you could replace this variable with a method `Game#even_round?` that determines if the round is even based on the `@sets`

File: lib/game.rb [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+  end
+
+  def end_game
+    winner = hwo_won
+
+    return @render.print_end_game_draw if winner == :draw
+
+    return @render.print_end_game(@player1, @player2) if winner == @player1
+
+    @render.print_end_game(@player2, @player1)
```
Comment (mapra99) [inline lib/game.rb]: What do you think about this?
```suggestion
    case @winner
    when :draw
      @render.print_end_game_draw
    when @player1
      @render.print_end_game(@player1, @player2)
    when @player2
      @render.print_end_game(@player2, @player1)
    end
```

**Review by mapra99 — COMMENTED** [review]
Nice work! 💯  Just left a few minor comments

File: lib/game.rb (hunk 9 above)
Comment (ImMPrada) [reply lib/game.rb]: and what do you think:

```
def end_game
  winner = who_won

  case winner
  when @player1
    @render.print_end_game(@player1, @player2)
  when @player2
    @render.print_end_game(@player2, @player1)
  else
    @render.print_end_game_draw
  end
end
```

File: lib/game.rb (hunk 8 above)
Comment (ImMPrada) [reply lib/game.rb]: > Also I think you could replace this variable with a method Game#even_round? that determines if the round is even based on the @sets

I think that too, I did it

File: lib/set.rb (hunk 7 above)
Comment (ImMPrada) [reply lib/set.rb]: Agree! i called it Round

File: lib/game.rb (hunk 9 above)
Comment (mapra99) [reply lib/game.rb]: much better ❤️

**Review by mapra99 — APPROVED** [review]
💯

**Pull request merged**